api/routes/websoc/lib.ts

```diff
@@ -0,0 +1,425 @@
+
+  if (parsedQuery.ge && parsedQuery.ge !== "ANY") {
+    AND.push({
+      geCategories: {
+        array_contains: parsedQuery.ge,
+      },
+    });
+  }
+
+  if (parsedQuery.department) {
+    AND.push({ department: parsedQuery.department });
+  }
+
+  if (parsedQuery.courseNumber) {
+    OR.push(
+      ...parsedQuery.courseNumber.map((n) =>
+        n.includes("-")
+          ? {
+              courseNumeric: {
+                gte: parseInt(n.split("-")[0].replace(/\D/g, "")),
+                lte: parseInt(n.split("-")[1].replace(/\D/g, "")),
+              },
+            }
+          : { courseNumber: n }
+      )
+    );
+  }
+
+  if (parsedQuery.instructorName) {
+    AND.push({
+      instructors: {
+        every: {
+          name: {
+            contains: parsedQuery.instructorName,
+          },
+        },
+      },
+    });
+  }
+
+  if (parsedQuery.courseTitle) {
+    AND.push({
+      courseTitle: {
+        contains: parsedQuery.courseTitle,
+      },
+    });
+  }
+
+  if (parsedQuery.sectionType && parsedQuery.sectionType !== "ANY") {
+    AND.push({ sectionType: parsedQuery.sectionType });
+  }
+
+  if (parsedQuery.startTime) {
+    AND.push({
+      meetings: {
+        every: {
+          startTime: {
+            gte: minutesSinceMidnight(parsedQuery.startTime),
+          },
+        },
+      },
+    });
+  }
+
+  if (parsedQuery.endTime) {
+    AND.push({
+      meetings: {
+        every: {
+          endTime: {
+            lte: minutesSinceMidnight(parsedQuery.endTime),
+          },
+        },
+      },
+    });
+  }
+
+  if (parsedQuery.division && parsedQuery.division !== "ANY") {
+    switch (parsedQuery.division) {
+      case "Graduate":
+        AND.push({ courseNumeric: { gte: 200 } });
+        break;
+      case "UpperDiv":
+        AND.push({ courseNumeric: { gte: 100, lte: 199 } });
+        break;
+      case "LowerDiv":
+        AND.push({ courseNumeric: { gte: 0, lte: 99 } });
+    }
+  }
+
+  if (parsedQuery.days) {
+    AND.push(
+      ...["Su", "M", "Tu", "W", "Th", "F", "Sa"]
+        .filter((x) => parsedQuery.days?.includes(x))
+        .map((x) => ({
+          meetings: {
+            every: {
+              days: {
+                array_contains: x,
+              },
+            },
+          },
+        }))
+    );
+  }
+
+  if (parsedQuery.fullCourses && parsedQuery.fullCourses !== "ANY") {
+    switch (parsedQuery.fullCourses) {
+      case "FullOnly":
+        AND.push({ sectionFull: true, waitlistFull: true });
+        break;
+      case "OverEnrolled":
+        AND.push({ overEnrolled: true });
+        break;
+      case "SkipFull":
+        AND.push({ sectionFull: true, waitlistFull: false });
+        break;
+      case "SkipFullWaitlist":
+        AND.push({ sectionFull: false, waitlistFull: false });
+    }
+  }
+
+  switch (parsedQuery.cancelledCourses) {
+    case undefined:
+    case "Exclude":
+      AND.push({ cancelled: false });
+      break;
```

Comment:
```suggestion
    case undefined:
       AND.push({ cancelled: false });
      break;
    case "Exclude":
      AND.push({ cancelled: false });
      break;
```
I get what it's supposed to do, but I think it's nice to have defined behavior for each case, even if they're duplicated.